dify/weaviate_retrieval.py

```diff
@@ -148,73 +208,257 @@ def retrieval(
+    )
+
+    # Validate authorization.
+    validate_authorization(authorization)
+
+    # Connect to Weaviate.
+    with closing(weaviate.connect_to_local()) as client:
+        # Get collection.
+        collection = get_weaviate_collection(client, request.knowledge_id)
 
-        # Embed the query
+        # Generate query embedding.
         try:
             vector = embed_with_ollama(request.query)
         except Exception as e:
+            _LOG.error("Embedding generation failed: %s", e)
             raise HTTPException(
                 status_code=500,
                 detail={"error_code": 5000, "error_msg": f"Embedding error: {e}"},
             )
 
-        # Perform vector search
-        try:
-            result = collection.query.near_vector(
-                near_vector=vector,
-                limit=request.retrieval_setting.top_k,
-                return_metadata=MetadataQuery(distance=True),
-            )
-        except Exception as e:
-            raise HTTPException(
-                status_code=500,
-                detail={
-                    "error_code": 5001,
-                    "error_msg": f"Weaviate query error: {e}",
-                },
-            )
+        # Perform vector search.
+        search_results = perform_vector_search(
+            collection, vector, top_k=request.retrieval_setting.top_k
+        )
 
-        # Build response records
-        records: list[dict] = []
-        for obj in result.objects:
-            # distance is now on obj.metadata.distance
-            distance = obj.metadata.distance or 0.0
-            score = max(0.0, 1.0 - distance)
-            if score < request.retrieval_setting.score_threshold:
-                continue
-
-            records.append(
-                {
-                    "metadata": {"filepath": obj.properties.get("filepath", "")},
-                    "score": score,
-                    "title": obj.properties.get(
-                        "filename", obj.properties.get("title", "")
-                    ),
-                    "content": obj.properties.get("text", ""),
-                }
-            )
+        # Build response records.
+        records = build_response_records(
+            search_results,
+            score_threshold=request.retrieval_setting.score_threshold,
+        )
+
+        _LOG.info(
+            "Retrieval completed successfully, returning %s records", len(records)
+        )
         return {"records": records}
 
 
-# Entry point
-if __name__ == "__main__":
+@app.get("/health")
+def health_check() -> Dict[str, str]:
+    """
+    Health check endpoint.
+
+    :return: health status
+    """
+    return {"status": "healthy", "service": "weaviate-retrieval"}
+
+
+# #############################################################################
+# Configuration and Startup.
+# #############################################################################
+
+
+def configure_logging(
```

Comment:
Can you use this as template ./helpers_root/dev_scripts_helpers/coding_tools/script_template.py ?